### PR TITLE
Trainer calibration button

### PR DESCRIPTION
**Pull request by dlktdr**
Solution for issue #662 

Will add another commit with auto-formatting.... once I figure out how it works right in VScode doesn't seem to want to use the clang file.. ;)

### CODE REVIEW

File: radio/src/gui/colorlcd/radio_trainer.cpp [hunk 1]
```diff
@@ -29,21 +29,30 @@ RadioTrainerPage::RadioTrainerPage():
...
+
+  void paint(BitmapBuffer * dc) {
+    auto curval = _getValue();
+    char buf[20];
+    if(textFlags & PREC1)
+      snprintf(buf, sizeof(buf), "%d.%.1d", curval / 10, abs(curval % 10));
+    else if(textFlags & PREC2)
+      snprintf(buf, sizeof(buf), "%d.%.2d", curval / 100, abs(curval % 100));
+    else
+      snprintf(buf, sizeof(buf), "%d", curval);
```
Comment (dlktdr) [inline radio/src/gui/colorlcd/radio_trainer.cpp:49]: I hope this is what was intended by the PREC flags.. value at -704 for instance with PREC1 should display as -70.4 PREC2 should be -7.04?

**Comment by dlktdr** [comment]
image

**Comment by pfeerick** [comment]
If `C_Cpp.clang_format_style` is set to file, it should be... otherwise I set the fallback to Google since it's mostly their style anyway.

**Comment by zyren** [comment]
Among the basic functions, only this part is what I am most looking forward to. The previous 4 digital boxes will be stuck if they are not operated properly, and the battery must be pulled out.
024BA3F9

File: radio/src/gui/colorlcd/radio_trainer.cpp (hunk 1 above)
Comment (pfeerick) [reply radio/src/gui/colorlcd/radio_trainer.cpp:49]: That's correct, isn't it @eshifri ?

**Comment by pfeerick** [comment]
LGTM... can calibrate a slave, and see the values of the sticks to help figure out if something is reversed or wrong with the slave. Had some fun getting my TX16S (x2) to behave as I only have stereo trainer leads, and it doesn't seem to like them - so half-plugged it is! TX12 didn't co-operate either, so it's probably the same.

**Comment by dlktdr** [comment]
> LGTM... can calibrate a slave, and see the values of the sticks to help figure out if something is reversed or wrong with the slave. Had some fun getting my TX16S (x2) to behave as I only have stereo trainer leads, and it doesn't seem to like them - so half-plugged it is! TX12 didn't co-operate either, so it's probably the same.

Yeah have to do that with mine too.. Just a little out and it works great ;)

**Pull request merged**